### ISSUE TITLE
Refactor Datatable in Accordance with W3C Authoring Practices Guidelines

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#4761]
closes #3740 
closes #3741 
closes #3805 
closes #4225 

The current implementation of our Datatable has serious usability/accessibility issues for both screen readers and keyboard users. A refactor in accordance with the [W3C Authoring Practices Guidelines 1.2 example for a Data Grid](https://www.w3.org/TR/2019/NOTE-wai-aria-practices-1.1-20190814/examples/grid/dataGrids.html) is the the way forward here.

Worth noting is that this implementation and refactor would not include _advanced Data Grid features (Excel/spreadsheet like funcntionality)_. We'd need to figure out some kind of strategy for teams to opt-in to those advanced (and potentially less accessible) features.

**Comment by joshblack** [comment]
@dakahn what are the features of data table that we're currently finding are inaccessible?
